components/brave_wallet/browser/brave_wallet_provider_impl.cc

```diff
@@ -444,24 +440,40 @@ void BraveWalletProviderImpl::SignTypedMessage(
     }
   }
 
+  if (domain_hash.empty() || primary_hash.empty()) {
+    std::move(callback).Run(
+        "", mojom::ProviderError::kInvalidParams,
+        l10n_util::GetStringUTF8(IDS_WALLET_INVALID_PARAMETERS));
+    return;
+  }
+  auto message_to_sign = EthSignTypedDataHelper::GetTypedDataMessageToSign(
+      domain_hash, primary_hash);
+  if (!message_to_sign || message_to_sign->size() != 32) {
+    std::move(callback).Run(
+        "", mojom::ProviderError::kInvalidParams,
+        l10n_util::GetStringUTF8(IDS_WALLET_INVALID_PARAMETERS));
+    return;
+  }
+
   // Convert to checksum address
   auto checksum_address = EthAddress::FromHex(address);
   GetAllowedAccounts(
-      false, base::BindOnce(&BraveWalletProviderImpl::ContinueSignMessage,
-                            weak_factory_.GetWeakPtr(),
-                            checksum_address.ToChecksumAddress(), message,
-                            std::move(eip712_hash), std::move(callback), true,
-                            domain_hash, primary_hash));
+      false,
+      base::BindOnce(&BraveWalletProviderImpl::ContinueSignMessage,
+                     weak_factory_.GetWeakPtr(),
+                     checksum_address.ToChecksumAddress(), message,
+                     std::move(*message_to_sign), base::HexEncode(domain_hash),
+                     base::HexEncode(primary_hash), true, std::move(callback)));
 }
 
 void BraveWalletProviderImpl::ContinueSignMessage(
     const std::string& address,
     const std::string& message,
-    std::vector<uint8_t>&& message_to_sign,
-    SignMessageCallback callback,
+    const std::vector<uint8_t>& message_to_sign,
```

Comment:
do not change this interface, it is designated for move only

Reply:
done